posthog/models/file_system/file_system_representation.py

```diff
@@ -0,0 +1,26 @@
+# posthog/models/file_system/file_system_representation.py
+
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass
+class FileSystemRepresentation:
+    """
+    Defines the minimal data needed to create/update a FileSystem entry:
+      - base_folder
+      - type
+      - ref
+      - name
+      - href
+      - meta
+      - should_delete: if True, we remove the entry instead of creating/updating
+    """
+
+    base_folder: str
+    type: str
+    ref: str
+    name: str
```

Comment:
feels like some of these could be auto-filled if `instance` was provided and e.g. `name` is not then you could `get_attr(instance, "name", "Untitled")` (order of args from memory 🙈 )

but i can see the argument for keeping them explicit too